src/components/mode-tools/mode-tools.jsx

```diff
@@ -33,185 +37,446 @@ import straightPointIcon from './icons/straight-point.svg';
-                        disabled={!(props.clipboardItems.length > 0)}
-                        imgSrc={pasteIcon}
-                        title={props.intl.formatMessage(messages.paste)}
-                        onClick={props.onPasteFromClipboard}
-                    />
-                </InputGroup>
-                <InputGroup className={classNames(styles.modLabeledIconHeight)}>
-                    <LabeledIconButton
-                        imgSrc={flipHorizontalIcon}
-                        title={props.intl.formatMessage(messages.flipHorizontal)}
-                        onClick={props.onFlipHorizontal}
-                    />
-                    <LabeledIconButton
-                        imgSrc={flipVerticalIcon}
-                        title={props.intl.formatMessage(messages.flipVertical)}
-                        onClick={props.onFlipVertical}
+                        disabled={!this.props.hasSelectedUnpointedPoints}
+                        imgSrc={straightPointIcon}
+                        title={this.props.intl.formatMessage(messages.pointed)}
+                        onClick={this.props.onPointPoints}
                     />
+                </div>
+            );
+        case Modes.SELECT:
+            return (
+                <div className={classNames(this.props.className, styles.modeTools)}>
+                    <InputGroup className={classNames(styles.modDashedBorder, styles.modLabeledIconHeight)}>
+                        <LabeledIconButton
+                            disabled={!this.props.selectedItems.length}
+                            imgSrc={copyIcon}
+                            title={this.props.intl.formatMessage(messages.copy)}
+                            onClick={this.props.onCopyToClipboard}
+                        />
+                        <LabeledIconButton
+                            disabled={!(this.props.clipboardItems.length > 0)}
+                            imgSrc={pasteIcon}
+                            title={this.props.intl.formatMessage(messages.paste)}
+                            onClick={this.props.onPasteFromClipboard}
+                        />
+                    </InputGroup>
+                    <InputGroup className={classNames(styles.modLabeledIconHeight)}>
+                        <LabeledIconButton
+                            imgSrc={flipHorizontalIcon}
+                            title={this.props.intl.formatMessage(messages.flipHorizontal)}
+                            onClick={this.props.onFlipHorizontal}
+                        />
+                        <LabeledIconButton
+                            imgSrc={flipVerticalIcon}
+                            title={this.props.intl.formatMessage(messages.flipVertical)}
+                            onClick={this.props.onFlipVertical}
+                        />
+                    </InputGroup>
+                </div>
+            );
+        case Modes.TEXT:
+            return (
+                <InputGroup>
```

Comment:
It looks like the other mode tools are wrapped in a `<div className={classNames(this.props.className, styles.modeTools)}>`, but the text input is not. This seems to be causing the slight "jump" when you choose the font tool:

![jumping-font-tool](https://user-images.githubusercontent.com/654102/40118989-78b07abc-58e9-11e8-8c8a-47534f2e62ed.gif)

Wrapping this InputGroup like the other tools fixes that.

Reply:
Good catch, thanks!